### PR TITLE
[Accessibility] Use product name as fallback alt text in product gallery

### LINKED ISSUE #48388: [Enhancement]: Improve Anchor Tag Accessibility in WooCommerce
brave_fBkCGsg7AP

**Issue:** Google PageSpeed Insights reports a deduction due to anchor tags lacking title or aria-label attributes, impacting accessibility for users relying on assistive technologies.

Guideline Reference: [Deque University link-name rule](https://dequeuniversity.com/rules/axe/4.9/link-name) emphasizes the need for discernible, unique, and focusable link text to improve navigation for screen reader users.

### Proposed Solution:
Add aria-label or title Attributes:
Ensure all anchor tags include meaningful aria-label or title attributes for better context and accessibility.

### Example:
**Current Code:**

`<a href="/example">Read More</a>`

**Proposed Code:**

`<a href="/example" aria-label="Read more about Example Product">Read More</a>`

Benefits:

- Improved Accessibility: Better experience for users with assistive technologies.
- Enhanced SEO: Improved semantic markup.
- Compliance: Aligns with accessibility standards.

### Describe alternatives you've considered

_No response_

### Additional context

This is the part of wc-template-functions.php file need to fix

```
function wc_get_gallery_image_html( $attachment_id, $main_image = false ) {
	$flexslider        = (bool) apply_filters( 'woocommerce_single_product_flexslider_enabled', get_theme_support( 'wc-product-gallery-slider' ) );
	$gallery_thumbnail = wc_get_image_size( 'gallery_thumbnail' );
	$thumbnail_size    = apply_filters( 'woocommerce_gallery_thumbnail_size', array( $gallery_thumbnail['width'], $gallery_thumbnail['height'] ) );
	$image_size        = apply_filters( 'woocommerce_gallery_image_size', $flexslider || $main_image ? 'woocommerce_single' : $thumbnail_size );
	$full_size         = apply_filters( 'woocommerce_gallery_full_size', apply_filters( 'woocommerce_product_thumbnails_large_size', 'full' ) );
	$thumbnail_src     = wp_get_attachment_image_src( $attachment_id, $thumbnail_size );
	$full_src          = wp_get_attachment_image_src( $attachment_id, $full_size );
	$alt_text          = trim( wp_strip_all_tags( get_post_meta( $attachment_id, '_wp_attachment_image_alt', true ) ) );
	$image             = wp_get_attachment_image(
		$attachment_id,
		$image_size,
		false,
		apply_filters(
			'woocommerce_gallery_image_html_attachment_image_params',
			array(
				'title'                   => _wp_specialchars( get_post_field( 'post_title', $attachment_id ), ENT_QUOTES, 'UTF-8', true ),
				'data-caption'            => _wp_specialchars( get_post_field( 'post_excerpt', $attachment_id ), ENT_QUOTES, 'UTF-8', true ),
				'data-src'                => esc_url( $full_src[0] ),
				'data-large_image'        => esc_url( $full_src[0] ),
				'data-large_image_width'  => esc_attr( $full_src[1] ),
				'data-large_image_height' => esc_attr( $full_src[2] ),
				'class'                   => esc_attr( $main_image ? 'wp-post-image' : '' ),
			),
			$attachment_id,
			$image_size,
			$main_image
		)
	);

	return '<div data-thumb="' . esc_url( $thumbnail_src[0] ) . '" data-thumb-alt="' . esc_attr( $alt_text ) . '" class="woocommerce-product-gallery__image"><a href="' . esc_url( $full_src[0] ) . '">' . $image . '</a></div>';
}
```

### LINKED ISSUE #51904: Accessibility: Empty Link [Product Page]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

If the shop owner does not provide alternative text the link to view the fullsized image is empty.

Image

### Component or Page: 

Page: Product Single (Simple)  

### Name affected element:

Image link, 'Intelligent Paper Knife'

### Relevant code snippets:

`<a href="https://woo.equalizedigital.com/wp-content/uploads/2024/08/img-ut.png"></a>`

### Recommended fix:

Add either an aria-label or fall back alternative text to the image to clarify the purpose of the link. 

"View image for product title."

### PR DESCRIPTION
**Pull request by Manussakis**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #48388 .
Closes #51904 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. In the frontend, go to a product page with a gallery. If you used the sample data to create the store, the Hoodie product can be used for testing.
2. Inspect the gallery and verify the images inside the anchor tag have alt texts. The parent of the anchor tag that wraps the image should also have the same alt text in the `data-thumb-alt` attribute.
Screenshot 2024-10-05 at 16 28 37

    - If the image already has an alt text defined in the Media Library, verify this is the image's alt text in the frontend.
Screenshot 2024-10-05 at 16 23 07
    - Otherwise, the alt text should follow this format: "{Product name} - Image {Image position in the gallery}"
Screenshot 2024-10-05 at 16 32 25
3. In the frontend, go to a product page with only feature image (without gallery)
4. Verify the image's alt text is either the text defined on the Media Library or the product name, but without the number of the image position, since there is no gallery.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by Manussakis** [comment]
@dkotter This PR is ready for code review. Thanks!

File: plugins/woocommerce/includes/wc-template-functions.php [hunk 1]
```diff
@@ -1679,6 +1683,7 @@ function wc_get_gallery_image_html( $attachment_id, $main_image = false ) {
 			'data-large_image_width'  => esc_attr( $full_src[1] ),
 			'data-large_image_height' => esc_attr( $full_src[2] ),
 			'class'                   => esc_attr( $main_image ? 'wp-post-image' : '' ),
+			'alt'                     => $alt_text,
```
Comment (dkotter) [inline plugins/woocommerce/includes/wc-template-functions.php]: Do we want escaping around this, similar to the other attributes above?

Comment (Manussakis) [reply plugins/woocommerce/includes/wc-template-functions.php]: Good catch @dkotter , somehow I missed it. 
Please review this commit. Thanks!
- [Escape alt attribute](https://github.com/woocommerce/woocommerce/pull/51936/commits/34a65e8184b9276671d6630800dbbf81ec154dbd)

**Review by dkotter — APPROVED** [review]
@ankitguptaindia @qasumitbagthariya This is ready for QA

**Comment by qasumitbagthariya** [comment]
**QA Update:**
---

I have checked this PR in the `Manussakis:feature/48388` branch and changes are working as expected.


image

image

image

**Testing Environment** 

<Details>

- WordPress: 6.6
- Theme: TT4 1.2
- WooCommerce - WooCommerce - Dev version with this fix specific branch.
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS Ventura 13.3 
- Git Branch: Manussakis:feature/48388

</Details>

@Manussakis please remove [WIP] from the title and take this PR out of draft.

**Comment by Manussakis** [comment]
@vikrampm1 This PR is ready for review by the appropriate Woo team. Thanks!

**Comment by vikrampm1** [comment]
@louwie17 / @woocommerce/woo-fse this PR is now ready for your review!

**Review by senadir — APPROVED** [review]
Thank you!

**Comment by vikrampm1** [comment]
@senadir this is now ready to be merged by you / Woo. Thanks!

**Comment by jeffpaul** [comment]
@senadir reminder ping on any final reviews here and then merge/release for Woo

**Pull request merged**